components/Button.vue

```diff
@@ -1,55 +1,217 @@
+          quantity: product.qty,
+          sku: product.sku,
+          category: 'PHYSICAL_GOODS'
+        })
+      })
+      return products
+    },
+    getBillingAddress () {
+      return {
+        address_line_1: store.state.checkout.paymentDetails.streetAddress,
+        address_line_2: store.state.checkout.paymentDetails.apartmentNumber,
+        admin_area_1: store.state.checkout.paymentDetails.region_code,
+        admin_area_2: store.state.checkout.paymentDetails.city,
+        postal_code: store.state.checkout.paymentDetails.zipCode,
+        country_code: store.state.checkout.paymentDetails.country
+      }
+    },
+    getShippingAddress () {
+      return {
+        name: {
+          full_name: store.state.checkout.shippingDetails.firstName + ' ' + store.state.checkout.shippingDetails.lastName
+        },
+        address: {
+          address_line_1: store.state.checkout.shippingDetails.streetAddress,
+          address_line_2: store.state.checkout.shippingDetails.apartmentNumber,
+          admin_area_1: store.state.checkout.shippingDetails.region_code,
+          admin_area_2: store.state.checkout.shippingDetails.city,
+          postal_code: store.state.checkout.shippingDetails.zipCode,
+          country_code: store.state.checkout.shippingDetails.country
+        }
+      }
+    },
     getAmount () {
-      return [{ amount: { value: this.grandTotal, currency_code: this.currencyCode } }]
+      return {
+        breakdown: {
+          item_total: {
+            currency_code: this.currencyCode,
+            value: this.getSegmentTotal('subtotal')
+          },
+          shipping: {
+            currency_code: this.currencyCode,
+            value: this.getSegmentTotal('shipping')
+          },
+          discount: {
+            currency_code: this.currencyCode,
+            value: this.getSegmentTotal('discount')
+          },
+          tax_total: {
+            currency_code: this.currencyCode,
+            value: this.getSegmentTotal('tax')
+          }
+        },
+        value: this.getSegmentTotal('grand_total'),
+        currency_code: this.currencyCode
+      }
     },
-    createOrder (data, actions) {
-      return actions.order.create({
-        purchase_units: this.getAmount()
+    async createOrderNvp (data, actions) {
+      return store.dispatch('cart/syncTotals', {
+        methodsData: {
+          country: store.state.checkout.shippingDetails.country,
+          zipCode: store.state.checkout.shippingDetails.zipCode,
+          region: store.state.checkout.shippingDetails.region,
+          region_id: store.state.checkout.shippingDetails.regionId,
+          region_code: store.state.checkout.shippingDetails.regionCode,
+          method_code: store.state.checkout.shippingDetails.shippingMethod,
+          carrier_code: store.state.checkout.shippingDetails.shippingCarrier,
+          payment_method: null
+        },
+        forceServerSync: true
+      }).then(() => {
```

Comment:
@tonyisworking If we use `async` don't need use `then`, now simply use `await`
+ @collymore